### DISCUSSION TITLE
Games not launching

### DISCUSSION THREAD

**Discussion by vcuongvu** [OnionUI/Onion#922] [Q&A] [answered]
### Checklist

- [X] I am reporting a bug in the Onion OS (Note that Onion devs can’t fix inherited issues with RA Cores)
- [X] I have reviewed the [Onion Wiki](https://github.com/OnionUI/Onion/wiki) and did not find the answer
- [X] I have searched the existing (open and closed) issues to ensure I do not log a duplicate

### Onions OS Version

4.2.0

### Miyoo Firmware Version

202305052130

### Provide a clear and concise description of the issue

I just installed a fresh install of OnionOS.
I've kept all the bios files from the original SD card provided by Miyoo.
I've added some games in the roms folder.
I refreshed the games list and deleted the hidden files but everytime I launch a game it keeps crashing.
When I launch any game from any console it shows :
- Loading
- The screen turns black
- Loading...
And then back to the game list.

### Steps to Reproduce

Fresh install
Adding all bios files
Adding roms
Refresh games list
Delete hidden files
Launching games

### Provide any additional context or information that may help us investigate your issue

No other context

**Comment by souziago** [comment]
Same thing here.

**Comment by XK9274** [comment]
What happens if you start Retroarch from the Apps page and load content? 
Do other apps load from the Apps page? (Tweaks, etc)
Do you have logging enabled? If so can you send us the logs.

Logging is a hidden menu item.

To enable it:
> Open RetroArch from Apps
> Settings
> User Interface
> Menu Item Visibility
> Settings
> Turn on logging
> Go back to the first Settings page
> Enter logging
> Turn all options on
> Go back and save your configuration file

Reproduce the failure, then go to /.tmp_update/logs and post the RA log files

* Remember to turn logging off again when you're done.

**Comment by vcuongvu** [comment]
When I open the Retroarch app, it crashed just like the games.
It shows Loading, black screen and Loading... and comes back to the apps menu.

Tweaks and Theme changes work normally.

I have three log files :
- batmon => Empty
- install :

:: Check firmware
STATS
Install RA check: 1
Onion total: 3249
RetroArch total: 994
:: Backup system
:: Remove configs
:: Install Onion
   - Extract '/mnt/SDCARD/.tmp_update/onion.pak' (3249 files) into /mnt/SDCARD
:: Install RetroArch
   - Extract '/mnt/SDCARD/RetroArch/retroarch.pak' (994 files) into /mnt/SDCARD
:: Install configs (reset: 1)
[CHECK] checking archive: /mnt/SDCARD/Themes/3DS mini by MarsTaco.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/3DS mini by MarsTaco
[CHECK] checking archive: /mnt/SDCARD/Themes/Analogue by Aemiii91.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/Analogue by Aemiii91
[CHECK] checking archive: /mnt/SDCARD/Themes/CyberOnion (2-pack) by Aemiii91.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/CyberOnion [Atomic] by Aemiii91
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/CyberOnion by Aemiii91
[CHECK] checking archive: /mnt/SDCARD/Themes/GBMini by Kitsuvi.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/GBMini by Kitsuvi
[CHECK] checking archive: /mnt/SDCARD/Themes/Hippo Stock.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/Hippo Stock
[CHECK] checking archive: /mnt/SDCARD/Themes/M by tenlevels.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/M by tenlevels
[CHECK] checking archive: /mnt/SDCARD/Themes/NanoSwitch (2-pack) by Amdy.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/NanoSwitch Dark by Amdy
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/NanoSwitch by Amdy
[CHECK] checking archive: /mnt/SDCARD/Themes/ONION PS (4-pack) by hanessh4.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/ONION PS Dark Blue by hanessh4
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/ONION PS Purple Grey by hanessh4
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/ONION PS Dark Grey by hanessh4
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/ONION PS Blue by hanessh4
[CHECK] checking archive: /mnt/SDCARD/Themes/Onion Boy DX by PixelShift.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/Onion Boy DX by PixelShift
[CHECK] checking archive: /mnt/SDCARD/Themes/Onion Boy by PixelShift + Jeltron.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/Onion Boy by PixelShift + Jeltron
[CHECK] checking archive: /mnt/SDCARD/Themes/Onionboy HD by Jeltron.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/Onionboy HD by Jeltron
[CHECK] checking archive: /mnt/SDCARD/Themes/Oransje by evolve.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/Oransje by evolve
[CHECK] checking archive: /mnt/SDCARD/Themes/PAC-MIYOO by tenlevels.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/PAC-MIYOO by tenlevels
[CHECK] checking archive: /mnt/SDCARD/Themes/RPO by TheDewd.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/RPO by TheDewd
[CHECK] checking archive: /mnt/SDCARD/Themes/Super Onion Entertainment System by TheDewd.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/Super Onion Entertainment System by TheDewd
[CHECK] checking archive: /mnt/SDCARD/Themes/System 9 by injekim.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/System 9 by injekim
[CHECK] checking archive: /mnt/SDCARD/Themes/TechDweeb by TechDweeb.zip
  [EXTRACT] -> /mnt/SDCARD/Themes/.previews/TechDweeb by TechDweeb
MI_SYS_Init cost 2066.000000ms
7779.000000 driver_name (nil)
7780.000000 FB_CreateDevice GFX_Open return 0
init audio 20
FB_EnterGraphicsMode start
iotcl VT_GETSTATE start
iotcl VT_ACTIVATE start 1
iotcl VT_WAITACTIVE start
iotcl KDGKBMODE start
iotcl KDSKBMODE start 3
iotcl KDSETMODE start
iotcl VT_LOCKSWITCH start
FB_EnterGraphicsMode end
_gfx_alloc_surface 0x350d0, 0xb648d000, idx 0
FB_AllocHWSurface1 0x35060, 0x350d0, 0xb648d000, flags 10001, 640x480, idx 0
SStar_CheckHWBlit
FB_FreeHWSurface 0x35060, 0x350d0, 0xb648d000
_gfx_free_surface 0x350d0 , 0xb648d000
FB_LeaveGraphicsMode A
ioctl KDSETMODE KD_TEXT
ioctl KDSETMODE KD_TEXT 3
ioctl VT_UNLOCKSWITCH
FB_FreeHWSurface 0x341a0, 0xb6f55af0, (nil)
MI_SYS_Init cost 0.000000ms
156140.000000 driver_name (nil)
156141.000000 FB_CreateDevice GFX_Open return 0
init audio 30
SDL_AudioInit(null)
open /dev/dsp return 7
audio driver dsp availale
FB_EnterGraphicsMode start
iotcl VT_GETSTATE start
iotcl VT_ACTIVATE start 1
iotcl VT_WAITACTIVE start
iotcl KDGKBMODE start
iotcl KDSKBMODE start 3
iotcl KDSETMODE start
iotcl VT_LOCKSWITCH start
FB_EnterGraphicsMode end
[InitKeyShm]: create shm fail : File exists, size 3280
_gfx_alloc_surface 0x431a0, 0xb6486000, idx 0
FB_AllocHWSurface1 0x43130, 0x431a0, 0xb6486000, flags 10001, 640x480, idx 0
SDL_OpenAudio freq 48000, format 32784, channels 2, samples 4096
open /dev/dsp return 7
opened /dev/dsp audio_fd is 7, samples = 4096
mi_ao sample rate=48000
mi_ao sample rate new =48000
open audio true
SStar_CheckHWBlit
FB_FreeHWSurface 0x43130, 0x431a0, 0xb6486000
_gfx_free_surface 0x431a0 , 0xb6486000
[SDL_RunAudio] : Task exiting. (TID-1236773808)
close audio_fd is 7
FB_LeaveGraphicsMode A
ioctl KDSETMODE KD_TEXT
ioctl KDSETMODE KD_TEXT 3
ioctl VT_UNLOCKSWITCH
FB_FreeHWSurface 0x421a0, 0xb6fafaf0, (nil)
MI_SYS_Init cost 0.000000ms
171414.000000 driver_name (nil)
171415.000000 FB_CreateDevice GFX_Open return 0
init audio 20
FB_EnterGraphicsMode start
iotcl VT_GETSTATE start
iotcl VT_ACTIVATE start 1
iotcl VT_WAITACTIVE start
iotcl KDGKBMODE start
iotcl KDSKBMODE start 3
iotcl KDSETMODE start
iotcl VT_LOCKSWITCH start
FB_EnterGraphicsMode end
[InitKeyShm]: create shm fail : File exists, size 3280
_gfx_alloc_surface 0x6c0d0, 0xb64ee000, idx 0
FB_AllocHWSurface1 0x6c060, 0x6c0d0, 0xb64ee000, flags 10001, 640x480, idx 0
SStar_CheckHWBlit
FB_LeaveGraphicsMode A
ioctl KDSETMODE KD_TEXT
ioctl KDSETMODE KD_TEXT 3
ioctl VT_UNLOCKSWITCH
FB_FreeHWSurface 0x6b1a0, 0xb6fb6af0, (nil)
------------------------------------------ before
                 mma heap name        heap_base_cpu_bus_addr                        length               chunk_mgr_avail
                mma_heap_name0                      26700000                       1500000                        fb3000
chunk_mgr info:
              offset              length               avail
                   0             1500000              fb3000

each chunk info:
              offset              length           used_flag           task_name
                   0               20000                   1              CMDMEM
               20000                1000                   1       sys-logConfig
               21000              400000                   1           fb_device
              421000              12c000                   1            blitsrc0
              54d000              fb3000                   0                  NA
------------------------------------------ 
MMA baseaddr(tmp) : 06700000
MMA baseaddr(fix) : 06700000
Released blitsrc0 offset : 00421000 length : 0012C000
------------------------------------------ after
                 mma heap name        heap_base_cpu_bus_addr                        length               chunk_mgr_avail
                mma_heap_name0                      26700000                       1500000                       10df000
chunk_mgr info:
              offset              length               avail
                   0             1500000             10df000

each chunk info:
              offset              length           used_flag           task_name
                   0               20000                   1              CMDMEM
               20000                1000                   1       sys-logConfig
               21000              400000                   1           fb_device
              421000             10df000                   0                  NA
------------------------------------------ 
:: Cleanup
MI_SYS_Init cost 0.000000ms
209486.000000 driver_name (nil)
209486.000000 FB_CreateDevice GFX_Open return 0
init audio 20
FB_EnterGraphicsMode start
iotcl VT_GETSTATE start
iotcl VT_ACTIVATE start 1
iotcl VT_WAITACTIVE start
iotcl KDGKBMODE start
iotcl KDSKBMODE start 3
iotcl KDSETMODE start
iotcl VT_LOCKSWITCH start
FB_EnterGraphicsMode end
[InitKeyShm]: create shm fail : File exists, size 3280
_gfx_alloc_surface 0x350d0, 0xb6419000, idx 0
FB_AllocHWSurface1 0x35060, 0x350d0, 0xb6419000, flags 10001, 640x480, idx 0
SStar_CheckHWBlit
FB_FreeHWSurface 0x35060, 0x350d0, 0xb6419000
_gfx_free_surface 0x350d0 , 0xb6419000
FB_LeaveGraphicsMode A
ioctl KDSETMODE KD_TEXT
ioctl KDSETMODE KD_TEXT 3
ioctl VT_UNLOCKSWITCH
FB_FreeHWSurface 0x341a0, 0xb6ee1af0, (nil)

network :

Thu Jan  1 04:00:03 UTC 1970 Network Checker: Update networking
Thu Jan  1 04:00:03 UTC 1970 Telnet: Killing firmware telnetd process

**Comment by vcuongvu** [comment]
I actually ordered 2 Miyoo Mini Plus and both have the same 20230505 firmware.
Both reacted the same way with fresh installs.

**Comment by XK9274** [comment]
Ok, so RA is completely failing.

If you know how to use Telnet, can you post me the result of these commands please:

`stat /mnt/SDCARD/RetroArch/retroarch`
`LD_TRACE_LOADED_OBJECTS=1 /mnt/SDCARD/RetroArch/retroarch`
`/mnt/SDCARD/RetroArch/retroarch -v > /mnt/SDCARD/.tmp_update/logs/ralaunch.log 2>&1`

The last one will generate a logfile in `/mnt/SDCARD/.tmp_update/logs/ralaunch.log`

**Reply by vcuongvu** [reply]
/mnt/SDCARD # stat /mnt/SDCARD/RetroArch/retroarch
stat: can't stat '/mnt/SDCARD/RetroArch/retroarch': No such file or directory
/mnt/SDCARD # ls /mnt/SDCARD/RetroArch/
config.xml  np2.cfg

**Reply by vcuongvu** [reply]
It seems like there is nothing under the folder RetroArch

**Reply by Aemiii91** [reply]
What about:
- `ls -a /mnt/SDCARD`
- `ls /mnt/SDCARD/.tmp_update`

**Reply by vcuongvu** [reply]
/mnt/SDCARD # ls -a /mnt/SDCARD/
.                          Emu                        System Volume Information
..                         Icons                      Themes
.VolumeIcon.icns           Media                      autorun.inf
.ash_history               RApp                       cachefile
.tmp_update                RetroArch                  miyoo
8188fu.ko                  Roms                       update.log
App                        Saves
BIOS                       Screenshots

**Reply by vcuongvu** [reply]
/mnt/SDCARD # ls /mnt/SDCARD/.tmp_update/
bin            lib            proxy.sh       script
cmd_to_run.sh  logs           res            updater
config         onionVersion   runtime.sh

**Reply by Aemiii91** [reply]
Okay, looks like everything is there. Can you do `ls -la /mnt/SDCARD/RetroArch/.retroarch` as well?

**Reply by XK9274** [reply]
Can you try running:

`find . -name ".*" -print -o -print | sed -e 's;[^/]*/;|____;g;s;____|; |;g' > /mnt/SDCARD/directory_tree.txt`

and post it as an attachment please? (create a text file and drag it into the comment box)

Sorry i've edited this to save to file, it'll spam the terminal

**Reply by vcuongvu** [reply]
/mnt/SDCARD # stat /mnt/SDCARD/RetroArch/.retroarch/
  File: /mnt/SDCARD/RetroArch/.retroarch/
  Size: 16384           Blocks: 32         IO Block: 16384  directory
Device: b301h/45825d    Inode: 5245        Links: 5
Access: (0777/drwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 1980-01-01 00:00:00.000000000
Modify: 1980-01-01 00:00:00.000000000
Change: 1980-01-01 00:00:00.000000000

**Reply by vcuongvu** [reply]
/mnt/SDCARD # ls -la /mnt/SDCARD/RetroArch/.retroarch/
total 192
drwxrwxrwx    5 root     root         16384 Jan  1  1980 .
drwxrwxrwx    3 root     root         16384 Jan  1  1980 ..
drwxrwxrwx    2 root     root         16384 Jan  1  1980 PrBoom
drwxrwxrwx    3 root     root         16384 Jan  1  1980 cores
-rwxrwxrwx    1 root     root        104620 Jan  1  1980 retroarch.cfg
drwxrwxrwx    2 root     root         16384 Jan  1  1980 system

**Reply by Aemiii91** [reply]
You are definitely missing some stuff in the RetroArch folder 🤔

**Reply by vcuongvu** [reply]
How do I get the file out ? With ftp ?

**Reply by XK9274** [reply]
You can use FTP yes or plug your SD into your PC

**Reply by vcuongvu** [reply]
[directory_tree.txt](https://github.com/OnionUI/Onion/files/11776729/directory_tree.txt)

**Reply by vcuongvu** [reply]
I don't know if it has an impact but when I first set it up, I unchecked the RetroArch Shortcut

**Reply by XK9274** [reply, marked as answer]
Sorry tons of data to go through there, the symptoms and issues do point to the obvious lack of files in your RetroArch dir, . You said you've rebuilt since then too? 

Can you attempt a backup, SD format & fresh install but don't copy/change anything immediately after the install finishes & see if RA runs?

**Reply by vcuongvu** [reply]
Ok I'll try that, there's no need for backup, I don't have anything on the device yet.

**Reply by vcuongvu** [reply]
So I took the source from there :
https://github.com/OnionUI/Onion/releases/download/latest/Onion-v4.2.0-beta-dev-b321c9a.zip

Reinstall it and RA does launch.

**Reply by vcuongvu** [reply]
Changes the theme to number 7 and RA still launches

**Reply by vcuongvu** [reply]
Removed RA from apps and reinserted it, still works, I'll try adding games now

**Reply by vcuongvu** [reply]
Holy shit it works !
Thank you very much guys !

So if it can help you for your code :

When I reinstalled OnionOS, I removed every apps except Themes and Tweaks, in the emulators, I chose GB, GBC, GBA, Sega Genesis and PS1 and I had the problem.

My guess is maybe by removing RetroArch from the app list the first time, it doesn't initiate all the necessary files.

Anyways, Thanks a lot !

**Reply by XK9274** [reply]
That's great to hear & thanks for the feedback!

**Comment by vcuongvu** [comment]
Ok, I'll try in a bit, just a mention :
I tried to install all the apps and all the standalone emulators under Expert and the menu Expert never appears even after reboot.

**Reply by Aemiii91** [reply]
You have to enable it in Tweaks.

**Reply by vcuongvu** [reply]
Thanks so I tried PS1, GB, GBC, GBA and Mega Drive and only PS1 with Tekken 3 worked.
It still doesn't work with the default core.

**Reply by Aemiii91** [reply]
Really sounds like you're missing some essential files. Have you tried reinstalling?

**Reply by vcuongvu** [reply]
Yes multiple times

**Reply by Aemiii91** [reply]
Have you tried a different SD card?

**Reply by vcuongvu** [reply]
Yes two different MM+ and two different SD Cards

**Comment by Aemiii91** [comment]
> **Steps to Reproduce**
> Fresh install
> Adding all bios files
> Adding roms
> Refresh games list
> **Delete hidden files** <---
> Launching games

What do you mean by this?

**Reply by vcuongvu** [reply]
I meant "Remove OSX system files"
Under Tweaks

That's what I found was advised on reddit.

**Reply by vcuongvu** [reply]
/mnt/SDCARD # ls -la /mnt/SDCARD/RetroArch/
total 96
drwxrwxrwx    3 root     root         16384 Jan  1  1980 .
drwxrwxrwx   16 root     root         16384 Jun 16 22:15 ..
-rwxrwxrwx    1 root     root          2301 Jan  1  1980 .atari800.cfg
drwxrwxrwx    5 root     root         16384 Jan  1  1980 .retroarch
-rwxrwxrwx    1 root     root          1180 Jan  1  1980 config.xml
-rwxrwxrwx    1 root     root           850 Jan  1  1980 np2.cfg

There is a hidden file .retroarch.
Is it supposed to be hidden ?

**Comment by nspire8** [comment]
I've also been experiencing this issue. Fresh install on a new device, using same MIYOO firmware version. I've tried to re-install several times with different variations of apps installed.

Looking around it seems I'm missing a retroarch binary.

```bash
       │ File: App/RetroArch/launch.sh
   1   │ #!/bin/sh
   2   │ echo $0 $*
   3   │
   4   │ cd /mnt/SDCARD/RetroArch/
   5   │ HOME=/mnt/SDCARD/RetroArch/ ./retroarch -v
```

From what I understand there should be a  binary in this folder:
```bash
$ ls -a Retroarch
.
..
.atari800.cfg
.retroarch
config.xml
np2.cfg
```

I tried using 7zip to extract retroarch.pak in the directory. RetroArch loads but it can't see any cores. I tried using the package manager to reinstall cores but no luck.

**Reply by degreeszero** [reply]
this is very similar to my issue. did a fresh install and it appear like there is nothing there, no cores. it almost feels like a joke.

**Reply by nspire8** [reply]
Tried a fresh install with zero cores and the default apps. Retroarch installed correctly and everything works now.

**Reply by nilsnordmark** [reply]
I had same problem, seems like a regression in the later betas. New Miyoo Mini Plus, same firmware 20230505, fresh new SD card. Retroarch is missing files / won't boot. Luckily I had a earlier beta (Onion-v4.2.0-beta-dev-4c7e2db) on my harddrive and that worked flawlessly.

**Comment by JulianKlug** [comment]
Had the same issue, resolved it by doing the following:

![image](https://github.com/OnionUI/Onion/assets/8020367/ede9276d-cbea-4951-94e4-a4cdf58c4ea9)

**Reply by Phanturian** [reply]
The Remove OSX files option is no longer present in the latest 4.2 RC build. I made a discussion topic asking about it, but nobody responded.
